### PR TITLE
[skip ci] Few small PEP8 amendments to basic.py

### PR DESCRIPTION
**Pull request by jremes-foss**
Few small PEP8 amendments to basic.py
- E201
- E265
- E302
- E231

Thank you.

### CODE REVIEW

File: sympy/core/basic.py [hunk 1]
```diff
@@ -1037,7 +1037,7 @@ def xreplace(self, rule):
         >>> x, y, z = symbols('x y z')
         >>> (1 + x*y).xreplace({x: pi})
         pi*y + 1
-        >>> (1 + x*y).xreplace({x:pi, y:2})
+        >>> (1 + x*y).xreplace({x: pi, y: 2})
```
Comment (toolforger) [inline sympy/core/basic.py]: I don't know what PEP rule this is - which is it?

File: sympy/core/basic.py [hunk 2]
```diff
@@ -1066,7 +1066,7 @@ def xreplace(self, rule):
 
         Trying to replace x with an expression raises an error:
 
-        >>> Integral(x, (x, 1, 2*x)).xreplace({x: 2*y}) #doctest: +SKIP
+        >>> Integral(x, (x, 1, 2*x)).xreplace({x: 2*y}) # doctest: +SKIP
```
Comment (toolforger) [inline sympy/core/basic.py]: Please test that this still skips the doctest.

File: sympy/core/basic.py [hunk 3]
```diff
@@ -1501,13 +1502,13 @@ def doit(self, **hints):
            >>> (2*Integral(x, x)).doit()
            x**2
 
-           >>> (2*Integral(x, x)).doit(deep = False)
+           >>> (2*Integral(x, x)).doit(deep=False)
```
Comment (toolforger) [inline sympy/core/basic.py]: No blanks around `=` on keyword parameters. (Yeah it's counter-intuitive.)

File: sympy/core/basic.py (hunk 1 above)
Comment (Newman101) [inline sympy/core/basic.py]: Hi, this is E231. 

http://pep8.readthedocs.org/en/latest/intro.html#error-codes

Comment (toolforger) [inline sympy/core/basic.py]: Somehow this looks odd to me, but I'm not sure whether it's SymPy code that's consistently using `{x:pi, y:2}`, or whether I saw that elsewhere (which would then be irrelevant).
Do you have a way to find out how the dict displays in SymPy are commonly written? I.e. how many with a space after the colon, and how many without a space after the colon?

**Comment by hargup** [comment]
@certik  do we accept PEP only changes? I personally don't think they are desirable.

**Comment by toolforger** [comment]
What is our mission statement on PEP8? (I didn't find any relevant mention on docs.sympy.org
If it's "we want PEP8-conformant code", we should accept such changes.
If it's "we prefer PEP8 but don't really care if it isn't", we should not.

**Comment by certik** [comment]
@hargup, @toolforger There are good arguments for both sides (i.e. nicer code vs a bit messier history). What is worse though is spending too much time discussing this, it's just bike shedding, and wasting everybody's time. I would suggest each of you to just intuitively decide, do we want this PR in as it is, or not? And then let's move on.

I am +1 to merge it as it is. I think the changes improve the code and I think there is no issue with history, it's not a big deal to find out what one needs.

**Comment by toolforger** [comment]
I'm undecided about the trade-off, so I'm fine with either decision.

**Comment by moorepants** [comment]
+1

**Comment by hargup** [comment]
@certik discussions about `git history` are recurrent and they often come up when a contributor is asked to squash their commits. I think the best way to avoid discussions on such unuseful topic is draft a policy so that we can point to that. And to avoid bikeshedding in the policy discussion I think you should draft it with some other older member (@moorepants, @asmeurer maybe?) without the having a consensus from the  larger community.

**Comment by moorepants** [comment]
@hargup I like the idea of having something written down too. And I think you are correct in that a fully consensus based approach to drafting that "rule" will take forever. But, maybe something along the lines of this https://github.com/sympy/sympy/wiki/The-Zen-of-SymPy could help set the tone. Here are my thoughts on this topic:

- PEP8 is generally good and definitely better than nothing.
- It's less time consuming to default to PEP8 guidelines than to argue about the merits of each PEP8 rule or devise our own style guidelines.
- Code that is merged into SymPy is for the community, leave personal style and personal conventions at home.
- Gaining a new contributor outweighs gaining "perfect" code or a "perfect" git history.
- We'd rather merge in code with a poor git history than not merge it in at all.
- Don't block contributions that generally improve the code due to minor style issues.

**Comment by Sumith1896** [comment]
+1 to all points @moorepants.
I agree that full consensus need not(cannot) be achieved for drafting "rules".

**Comment by toolforger** [comment]
The issue with the Zen of SymPy is that it does not say "why".
That's a real problem, because people won't know when to bend or ignore 
a rule if they have reasons but don't know whether their reasons are 
better than those behind the Zen.

Though I do like the Zen format. It's just that the individual bullet 
points should have links to their respective rationales.

> Here are my thoughts on this topic:
>
> - PEP8 is generally good and definitely better than nothing.

Yes.

> - It's less time consuming to default to PEP8 guidelines than to argue about the merits of each PEP8 rule or devise our own style guidelines.

Yes.

However:

- More uniform coding style is better than less uniform coding style.
- PEP 8 leaves some coding style choices to the project. (Indentation 
for follow-up lines being the most prominent example.)

> - Code that is merged into SymPy is for the community, leave personal style and personal conventions at home.

Yes.

> - Gaining a new contributor outweighs gaining "perfect" code or a "perfect" git history.

Sort-of. I'd want to make it clear to new contributors what aspects 
could be improved.
I tend to term it as "this is good, I'd like this-and-that improved but 
I'm definitely not going to -1 your contribution for that reason". I 
found that most contributors *want* to have that fixed by themselves. 
Seems like it's perceived as a bonus goal or something (and I generally 
like it so I guess everybody is happy that way).

Maybe that's what makes pure PEP8 contributions a bit contentious: if 
the author misinterpreted something, this tends to devalue most if not 
all of a PR. (Juha has been very patient with that, retracting or 
reworking PRs. So now that I revisit that history, I can say that Juha 
is a good PEP8 improver. He's been placing the project's style 
guidelines above his own, to the point that I don't know if he even has 
any - which definitely isn't a bad thing for style improvement work.)

> - We'd rather merge in code with a poor git history than not merge it in at all.
> - Don't block contributions that generally improve the code due to minor style issues.

Definitely.

**Pull request merged**